server/src/capture.rs

```diff
@@ -0,0 +1,227 @@
+// src/capture.rs
+
+// ## Imports
+//
+// ### Standard library
+use std::fs;
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+
+// ### Third-party
+use chrono::Local;
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use tokio::sync::Mutex;
+use tokio_postgres::{Client, NoTls};
+
+// ### Local
+
+/*
+The `Event` struct represents an event to be stored in the database.
+
+Fields:
+    - `user_id`: The ID of the user associated with the event.
+    - `event_type`: The type of event (e.g., "keystroke", "file_open").
+    - `data`: Optional additional data associated with the event.
+
+# Example
+
+let event = Event {
+    user_id: "user123".to_string(),
+    event_type: "keystroke".to_string(),
+    data: Some("Pressed key A".to_string()),
+};
+
+ */
+#[derive(Deserialize, Debug)]
+pub struct Event {
+    pub user_id: String,
+    pub event_type: String,
+    //    pub timestamp: String,
+    pub data: Option<String>,
+}
+
+/* The `Config` struct represents the database connection parameters read from `config.json`.
```

Comment:
A singleton, correct (e.g. just one instance for the whole program)? If so, add this to the docs.

Reply:
Yes, the singleton design will be implemented in the next PR